src/Bicep.Core/Semantics/ImportedSymbol.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Semantics.Metadata;
+using Bicep.Core.Registry;
+using Bicep.Core.Syntax;
+using Bicep.Core.TypeSystem;
+
+namespace Bicep.Core.Semantics;
+
+public class ImportedSymbol : DeclaredSymbol
+{
+    private readonly Lazy<ISemanticModel?> sourceModelLazy;
+
+    public ImportedSymbol(ISymbolContext context, ImportedSymbolsListItemSyntax declaringSyntax, CompileTimeImportDeclarationSyntax enclosingDeclartion)
+        : base(context, declaringSyntax.Name.IdentifierName, declaringSyntax, declaringSyntax.Name)
+    {
+        EnclosingDeclaration = enclosingDeclartion;
+
+        sourceModelLazy = new(() =>
+        {
+            SemanticModelHelper.TryGetSemanticModelForForeignTemplateReference(Context.Compilation.SourceFileGrouping,
+                EnclosingDeclaration,
+                b => b.CompileTimeImportDeclarationMustReferenceTemplate(),
+                Context.Compilation,
+                out var semanticModel,
+                out _);
+
+            return semanticModel;
+        });
+    }
+
+    public CompileTimeImportDeclarationSyntax EnclosingDeclaration { get; }
+
+    public ImportedSymbolsListItemSyntax DeclaringImportedSymbolsListItem => (ImportedSymbolsListItemSyntax)DeclaringSyntax;
+
+    public string? OriginalSymbolName => DeclaringImportedSymbolsListItem.OriginalSymbolName switch
+    {
+        IdentifierSyntax identifier => identifier.IdentifierName,
+        StringSyntax @string when Context.TypeManager.GetTypeInfo(@string) is StringLiteralType stringLiteral => stringLiteral.RawStringValue,
```

Comment:
Involving type checking feels a bit round-about. Why not just fetch this directly from the syntax?

Reply:
The type manager is the only place in the code base where we currently perform constant folding, so checks like this are sprinkled pretty liberally throughout the compiler. (I could add a `TryFoldConstant(<syntax>)` method, but it would just be an indirect way to query the type system.) 

Getting the value directly from the syntax will work if interpolation is blocked.